src/Microsoft.AspNetCore.SystemWebAdapters.Authentication/RemoteAuthenticationResultFactory.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+/// <summary>
+/// Factory type for generating remote authentication results from HTTP responses
+/// from a remote app.
+/// </summary>
+public class RemoteAuthenticationResultFactory : IAuthenticationResultFactory
+{
+    private readonly ILogger<RemoteAuthenticationResultFactory> _logger;
+
+    public RemoteAuthenticationResultFactory(ILogger<RemoteAuthenticationResultFactory> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task<RemoteAuthenticationResult> CreateRemoteAuthenticationResultAsync(HttpResponseMessage response, RemoteAuthenticationOptions options)
+    {
+        RemoteAuthenticationResult? ret = null;
+
+        // If the result has a 200 status code, attempt to deserialize the ClaimsPrincipal
+        if (response.StatusCode == HttpStatusCode.OK)
+        {
+            try
+            {
+                using var reader = new BinaryReader(await response.Content.ReadAsStreamAsync());
+                ret = new RemoteAuthenticationResult(new ClaimsPrincipal(reader), (int)response.StatusCode);
+            }
+            catch (Exception exc)
+            {
+                _logger.LogWarning(exc, "Failed to deserialize remote authentication response as a claims principal despite success status code");
```

Comment:
Do we want to silently ignore? Or do we want to log and throw?

Reply:
I think we don't need to throw here. Even if there's no ClaimsPrincipal, we can still report the status code and headers. Likely something's wrong on the server side, but I lean towards allowing the client to still work (just without remote auth) and providing log messages.

If it turns out that this comes up a lot and we need to make the failure more obvious, we could switch to throwing.